crates/re_renderer/src/point_cloud_builder.rs

```diff
@@ -194,79 +191,65 @@ impl<'a> PointCloudBatchBuilder<'a> {
         self.batch_mut().point_count += num_points as u32;
 
         {
-            re_tracing::profile_scope!("positions");
-            let num_before = self.0.vertices.len();
+            re_tracing::profile_scope!("positions & radii");
             self.0.vertices.extend(
-                positions
+                izip!(positions.iter().copied(), radii.iter().copied())
                     .take(num_points)
-                    .zip(radii.take(num_points))
-                    .map(|(position, radius)| PointCloudVertex { position, radius }),
-            );
-            // Fill up with defaults. Doing this in a separate step is faster than chaining the iterator.
-            let num_default = num_points - (self.0.vertices.len() - num_before);
-            self.0.vertices.extend(
-                std::iter::repeat(PointCloudVertex {
-                    position: glam::Vec3::ZERO,
-                    radius: Size::AUTO,
-                })
-                .take(num_default),
+                    .map(|(pos, radius)| PositionRadius { pos, radius }),
             );
         }
         {
             re_tracing::profile_scope!("colors");
-            if let Some(num_written) = self
-                .0
+
+            self.0
+                .color_buffer
+                .extend_from_slice(colors)
```

Comment:
color/radius/etc. slices should be limited to be max the size of the positions buffer